client/src/main/java/gui/printers/FieldPrinter.java

```diff
@@ -23,40 +31,83 @@ public static void insertFields(GridPane printingBoard) {
+    }
+
+    private static void setDimensions(Button field) {
+        field.setPrefHeight(FIELD_HEIGHT);
+        field.setPrefWidth(FIELD_WIDTH);
     }
 
     private static void addStyle(Button field) {
-        field.getStyleClass().add("water-button");
+        field.getStyleClass().add(WATER_STYLE);
     }
 
-    private static void addMissOnClick(Button field) {
+    private static void addShotOnAction(Button field, int widthCoordinates, int heightCoordinates) {
         ClientAppRunner appInstance = ClientAppRunner.getInstance();
         ServerConnection serverConnection = appInstance.getServerConnection();
-        field.setOnAction(event -> serverConnection.sendMessage(SHOT_COMMAND));
+        String coordinatesToPass = widthCoordinates + "-" + heightCoordinates;
+        field.setOnAction(event -> serverConnection.sendMessage(SHOT_COMMAND + ":" + coordinatesToPass));
     }
 
-    private static void setDimensions(Button field) {
-        field.setPrefHeight(FIELD_HEIGHT);
-        field.setPrefWidth(FIELD_WIDTH);
-    }
+    private static void addMockOnAction(Button field, int widthCoordinates, int heightCoordinates) {
+        Field fieldToChange = ShotResult.getShotInformation(ShotResult.outputSupplier());
+        Coordinates coordinatesOfField = Coordinates.create(widthCoordinates, heightCoordinates);
+        FieldState stateOfField = fieldToChange.getCoordinateState(coordinatesOfField);
+        changeColorFromState(stateOfField, field);
 
+    }
 
-    private static void addToEachSpace(int numberOfRows, int numberOfColumns, GridPane printingBoard) {
-        for (int widthPosition = 0; widthPosition < numberOfRows; widthPosition++) {
-            for (int heightPosition = 0; heightPosition < numberOfColumns; heightPosition++) {
-                Button fieldToClick = createField();
-                printingBoard.add(fieldToClick, widthPosition, heightPosition);
-            }
+    private static void changeColorFromState(FieldState state, Button button) {
+        if (null == state) {
+            return;
         }
+        switch (state) {
+            case SEEN:
+                addSeenMark(button);
+                break;
+            case DAMAGED:
+                addDamagedMark(button);
+                break;
+            case DESTROYED:
+                addDestroyedMark(button);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private static void addSeenMark(Button button) {
+        button.setText("O");
+    }
+
+    private static void addDamagedMark(Button button) {
+        ObservableList<String> styles = button.getStyleClass();
+        styles.removeAll(WATER_STYLE);
+        styles.add(DAMAGED_STYLE);
+    }
+
+    private static void addDestroyedMark(Button button) {
+        ObservableList<String> styles = button.getStyleClass();
+        styles.removeAll(WATER_STYLE);
+        styles.add(DESTROYED_STYLE);
```

Comment:
No matter what is done in class 100 LoC is screaming, I am not SRP. (ofcourse there may be exceptions).

LoC: Line of Code

Here: Strategy pattern will help

Reply:
Will do in the future. Tracking in #35